common/src/main/java/com/scottlogic/deg/common/profile/ProfileFields.java

```diff
@@ -50,4 +50,8 @@ public boolean equals(Object obj) {
     public int hashCode() {
         return fields.hashCode();
     }
+
+    public List<Field> getFields() {
```

Comment:
still not conviced this is needed :p

Reply:
Which would you prefer? The `getFields()` method with:
```
for (Field field : tree.getFields().getFields()) {
    fieldSpecs.put(field, FieldSpec.Empty);
}
```
Or without the method:
```
for (Field field : tree.getFields().stream().collect(Collectors.toList())) {
    fieldSpecs.put(field, FieldSpec.Empty);
}
```
Or without the method:
```
tree.getFields().iterator().forEachRemaining(field -> fieldSpecs.put(field, FieldSpec.Empty));
```

Reply:
the implementation of getfields is correct, just there's nothing inherrently wrong with public variables on a data class. so there's not much reason for this extra method. 

you don't need to mock the decisiontree, it's a data class